src/Umbraco.Web.Common/Controllers/RenderController.cs

```diff
@@ -37,7 +38,17 @@ public RenderController(ILogger<RenderController> logger, ICompositeViewEngine c
         /// <summary>
         /// Gets the umbraco context
         /// </summary>
-        protected IUmbracoContext UmbracoContext => _umbracoContextAccessor.UmbracoContext;
+        protected IUmbracoContext UmbracoContext
+        {
+            get
+            {
+                if (!_umbracoContextAccessor.TryGetUmbracoContext(out var umbracoContext))
+                {
+                    throw new InvalidOperationException("Wasn't able to get an UmbracoContext");
+                }
+                return umbracoContext;
```

Comment:
Here you can just return `_umbracoContextAccessor.GetRequiredUmbracoContext();` after adding the `using Umbraco.Extensions;`

This is basically a similar case to the code duplication comment that @bergmania made but for the umbracoContext, as there are many places where we are doing this check